sw/device/lib/dif/dif_spi_device.h

```diff
@@ -0,0 +1,65 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_DIF_SPI_DEVICE_H_
+#define SW_DEVICE_LIB_DIF_DIF_SPI_DEVICE_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/dif/register.h"
+
+/**
+ * Initializes the SPI device located at |base_addr|.
+ *
+ * This function *must* be called on a particular |reg32_t| before calling any
+ * other functions in this header with that |reg32_t|; moreover, it *may not*
+ * be called more than once with any |reg32_t|. All deviations are Undefined
+ * Behavior.
+ *
+ * @param base_addr the base address for the SPI device registers.
+ */
+void spi_device_init(reg32_t base_addr);
+
+/**
+ * Attempts to send |len| bytes from the buffer pointed to by |data| over the
+ * initialized SPI device at |base_addr|.
+ *
+ * |data| *must* point to a buffer of at least |len| bytes of initialized data;
+ * not doing so is Undefined Behavior.
+ *
+ * @param base_addr the base address for the SPI device registers.
+ * @param data a contiguous buffer to copy from.
+ * @param len the length of the buffer |data| points to.
+ * @return the number of bytes actually sent.
+ */
+size_t spi_device_send(reg32_t base_addr, const void *data, size_t len);
+
+/**
+ * Attempts to recieve |len| bytes over the SPI device at |base_addr|, which
+ * will be written to the buffer pointed to by |data|.
+ *
+ * |data| *must* point to a buffer of at least |len| bytes of allocated data,
+ * which may be uninitialized; not doing so is Undefined Behavior.
+ *
+ * After this function returns, the first |len| bytes pointed to by |buf| will
+ * be in an initialized state. This function will never read from |buf|.
+ *
+ * @param base_addr the base address for the SPI device registers.
+ * @param buf a contiguous, possibly uninitialized buffer to write to.
+ * @param len the length of the buffer |data| points to.
+ * @return the number of bytes actually recieved.
+ */
+size_t spi_device_recv(reg32_t base_addr, void *buf, size_t len);
```

Comment:
I much rather prefer dif_spi_device_recv and similar - there can/will be higher level drivers that will utilise difs, and it will be much more easy to understand what is happening if DIF public functions are prefixed with dif_*

Reply:
Done. Is this acknowledged in Garret's doc?

Reply:
I have left a comment on the DIF doc, but I don't think it has been acknowledged yet. I just thought it's worth mentioning here and then we could discuss.